src/Arcus.Messaging.Pumps.Abstractions/Resiliency/DefaultMessagePumpCircuitBreaker.cs

```diff
@@ -40,70 +38,31 @@ public DefaultMessagePumpCircuitBreaker(IServiceProvider serviceProvider, ILogge
+                return Task.CompletedTask;
             }
 
             if (messagePump.CircuitState != MessagePumpCircuitState.Closed)
             {
-                _logger.LogWarning($"Cannot pause MessagePump for JobId {jobId} because the MessagePump's circuitbreaker is not in a closed state.");
-                return;
+                _logger.LogWarning("Cannot pause message pump '{JobId}' because the pump's circuit breaker is not in a closed state", jobId);
+                return Task.CompletedTask;
             }
 
-            _logger.LogDebug("Open circuit by pausing message processing for message pump '{JobId}'...", jobId);
-            await messagePump.StopProcessingMessagesAsync(CancellationToken.None);
-
-            await Task.Factory.StartNew(async () =>
-            {
-                await WaitRecoveryTimeAsync(messagePump, options);
-
-                MessageProcessingResult result;
-                do
-                {
-                    result = await TryProcessSingleMessageAsync(messagePump, options);
-
-                    if (!result.IsSuccessful)
-                    {
-                        await WaitMessageIntervalDuringRecoveryAsync(messagePump, options);
-                    }
-
-                } while (!result.IsSuccessful);
-
-                await ResumeMessageProcessingAsync(jobId);
-            }, TaskCreationOptions.LongRunning);
-        }
-
-        /// <summary>
-        /// Continue the process of receiving messages in the message pump after a successful message handling.
-        /// </summary>
-        /// <param name="jobId">The unique identifier to distinguish the message pump in the application services.</param>
-        /// <exception cref="ArgumentException">Thrown when the <paramref name="jobId"/> is blank.</exception>
-        public virtual async Task ResumeMessageProcessingAsync(string jobId)
-        {
-            Guard.NotNullOrWhitespace(jobId, nameof(jobId));
-
-            MessagePump messagePump = GetRegisteredMessagePump(jobId);
-
-            if (messagePump.IsStarted)
-            {
-                _logger.LogWarning("Resume called on Message pump '{JobId}' but Message pump is already started. CircuitState = {CircuitState}", jobId, messagePump.CircuitState);
-                return;
-            }
+            var options = new MessagePumpCircuitBreakerOptions();
+            configureOptions?.Invoke(options);
 
-            _logger.LogInformation("Message pump '{JobId}' successfully handled a single message, resume message processing (circuit breaker: closed)", messagePump.JobId);
-            await messagePump.StartProcessingMessagesAsync(CancellationToken.None);
+            _logger.LogDebug("Open circuit by pausing message processing for message pump '{JobId}'...", jobId);
 
+            messagePump.NotifyPauseReceiveMessages(options);
+            return Task.CompletedTask;
```

Comment:
I would still leave the `PauseMessageProcessingAsync` asynchronous as the `Default...` one is only our default implementation of an interfact, not the only possible one.